validations/src/tests.rs

```diff
@@ -3053,6 +3056,87 @@ fn test_block<F: FnMut(&mut Block) -> bool>(mut mut_block: F) -> Result<(), fail
     .map(|_| ())
```

Comment:
You can refactor the `test_block` function into:

    test_block_with_drpool(mut_block, DataRequestPool::default())

So we avoid some duplicated code